src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -6,7 +6,7 @@
 /**
  * Parameterize this service and add its implementation in a separate class.
  */
-public interface MachineService {
+public interface MachineService<T extends Machine> {
```

Comment:
Like Sofia said in the Q&A, we can parametrize MachineService with <T> and then use <? extends T> or <? super T>, so that we are not restricted by the class Machine

Reply:
thank for comments I will fix that